lib/oban.ex

```diff
@@ -840,4 +861,16 @@ defmodule Oban do
 
     with {:error, error} <- conf.engine.init(conf, opts), do: raise(error)
   end
+
+  defp maybe_validate_queue_exists!(name, opts) do
+    if Keyword.get(opts, :local_only) do
+      validate_queue_exists!(name, opts[:queue])
+    end
+  end
+
+  defp validate_queue_exists!(name, queue) do
+    unless is_pid(Registry.whereis(name, {:producer, to_string(queue)})) do
+      raise ArgumentError, "expected queue does not exist locally"
```

Comment:
It would be nice to include the queue name in the message:

```suggestion
      raise ArgumentError, "queue #{inspect(queue)} does not exist locally"
```